lib/charms/mysql/v0/mysql.py

```diff
@@ -1209,6 +1209,21 @@ def render_mysqld_configuration(  # noqa: C901
             config.write(string_io)
             return string_io.getvalue(), dict(config["mysqld"])
 
+    def _build_mysql_database_dba_role(self, database: str) -> str:
+        """Builds the database-level DBA role, given length constraints."""
+        role_prefix = "charmed_dba"
+        role_suffix = "XX"
+
+        role_name_available = ROLE_MAX_LENGTH - len(role_prefix) - len(role_suffix) - 2
+        role_name_description = database[:role_name_available]
+        role_name_collisions = self.list_mysql_roles(f"{role_prefix}_{role_name_description}_%")
+
+        return "_".join((
+            role_prefix,
+            role_name_description,
+            str(len(role_name_collisions)).zfill(len(role_suffix)),
```

Comment:
Situation: we have 3 roles charmed_dba_test_00, charmed_dba_test_01, charmed_dba_test_02.
1) When we drop the relation with `charmed_dba_test_00`, removal of `charmed_dba_test_00` should happen.
2) When we add a new relation, this code can try to recreate `charmed_dba_test_02` and exit with an error.